code/frameworks/sveltekit/src/preview.ts

```diff
@@ -0,0 +1,104 @@
+import type { Decorator } from '@storybook/svelte';
+
+import { onMount } from 'svelte';
+import { setAfterNavigateArgument } from './mocks/app/navigation';
+import { setNavigating, setPage, setUpdated } from './mocks/app/stores';
+
+export const decorators: Decorator[] = [
+  (Story, ctx) => {
+    setPage(ctx.parameters?.sveltekit?.stores?.page);
+    setUpdated(ctx.parameters?.sveltekit?.stores?.updated);
+    setNavigating(ctx.parameters?.sveltekit?.stores?.navigating);
+    setAfterNavigateArgument(ctx.parameters?.sveltekit?.navigation?.afterNavigate);
+
+    onMount(() => {
+      const globalClickListener = (e: MouseEvent) => {
+        // we add a global click event listener and we check if there's a link in the composedPath
+        const path = e.composedPath();
+        const hasLink = path.findLast((el) => el instanceof HTMLElement && el.tagName === 'A');
+        if (hasLink && hasLink instanceof HTMLAnchorElement) {
+          // if it has a link we get the href of the link and we check over every provided href using the
+          // key as a regex
+          const to = hasLink.getAttribute('href');
+          if (ctx?.parameters?.sveltekit?.hrefs && to) {
+            Object.entries(ctx.parameters.sveltekit.hrefs).forEach(([link, override]) => {
+              if (override instanceof Function) {
+                const regex = new RegExp(link);
+                if (regex.test(to)) {
+                  // if the regex pass we call the function the user provided
+                  override();
+                }
```

Comment:
This is going to bite us.

Users setting the key as a string (like you've done in the test stories) will expect exact matches, but it will actually be regexes because of this.

eg:
```ts
  parameters: {
    sveltekit: {
      hrefs: {
        '/my-weird.link': link,
      },
    },
  },
```

This will both match `"/my-weird.link"` _and_ `"/my-weird!link"` because the `.` is actually a regex wildcard. (it's a weird edge case, but someone is going to file an issue about this.

I think supporting regexes and strings are great, we should just do different logic based on the two.

Like

```suggestion
                if (typeof link === 'string'){
                  if (link === to){
                    override();
                  }
                } else if (link instanceof RegExp) {
                  if (link.test(to)){
                    override();
                  }
                } else {
                  throw new Error(`href overrides must either be strings or regular expressions, but got: ${typeof link}`);
                }
```

Thoughts?

Reply:
You can't use regex to index an object right? Maybe we should define a convenction? Like if the string starts with `rx:` it's a regex?